app/tests/archives_tests/test_views.py

```diff
@@ -444,6 +445,105 @@ def test_api_archive_item_add_and_update_non_image_file(
 
 
+@pytest.mark.django_db
+def test_api_archive_item_add_and_update_image_file(
+    client, settings, django_capture_on_commit_callbacks
+):
+    # Override the celery settings
+    settings.task_eager_propagates = (True,)
+    settings.task_always_eager = (True,)
+
+    archive = ArchiveFactory()
+    editor = UserFactory()
+    archive.add_editor(editor)
+    item = ArchiveItemFactory(archive=archive)
+    ci = ComponentInterfaceFactory(
+        kind=InterfaceKind.InterfaceKindChoices.IMAGE, store_in_database=False
+    )
+
+    def set_civ(data):
+        with django_capture_on_commit_callbacks() as callbacks:
+            response = get_view_for_user(
+                viewname="api:archives-item-detail",
+                reverse_kwargs={"pk": item.pk},
+                data=data,
+                user=editor,
+                client=client,
+                method=client.patch,
+                content_type="application/json",
+                HTTP_X_FORWARDED_PROTO="https",
+            )
+        recurse_callbacks(
+            callbacks=callbacks,
+            django_capture_on_commit_callbacks=django_capture_on_commit_callbacks,
+        )
+
+        assert response.status_code == 200
+        assert response.json()["pk"] == str(item.pk)
+
+        # Sanity
+        image = item.values.first().image
+        assert image.name == "image10x10x10.mha"
+        assert image.files.count() == 1
+
+    def set_civ_via_user_upload():
+        user_upload = create_upload_from_file(
+            creator=editor, file_path=RESOURCE_PATH / "image10x10x10.mha"
+        )
+        set_civ(
+            data={
+                "values": [
+                    {
+                        "interface": ci.slug,
+                        "user_upload": user_upload.api_url,
+                    }
+                ]
+            },
+        )
+
+    def set_civ_via_upload_session():
+        user_upload = create_upload_from_file(
+            creator=editor, file_path=RESOURCE_PATH / "image10x10x10.mha"
+        )
+
+        upload_session = RawImageUploadSessionFactory(creator=editor)
+        upload_session.user_uploads.set([user_upload])
+
+        set_civ(
+            data={
+                "values": [
+                    {
+                        "interface": ci.slug,
+                        "upload_session": upload_session.api_url,
+                    }
+                ]
+            },
+        )
+
+    # Add
+    assert item.values.count() == 0
+    set_civ_via_upload_session()
+    assert item.values.count() == 1
+    item.values.clear()
+
+    # Update
+    assert item.values.count() == 0
+    set_civ_via_upload_session()
+    assert item.values.count() == 1
+    item.values.clear()
+
+    # Add
+    assert item.values.count() == 0
+    set_civ_via_user_upload()
+    assert item.values.count() == 1
+    item.values.clear()
+
+    # Update
+    assert item.values.count() == 0
+    set_civ_via_user_upload()
+    assert item.values.count() == 1
```

Comment:
What is the difference between add and update here? You're calling the exact same function and clear all values before, so same starting point and same api call, right?

Reply:
Ah, ehm. Good point! It SHOULD update. I now see it does not because I am clearing things out. That should only happen at the start.